### DISCUSSION TITLE
How can I update xmake and xmake-repo by SSH instead of HTTPS?

### DISCUSSION THREAD

**Discussion by xq114** [xmake-io/xmake#1839] [Q&A]
目前xmake update -s和xrepo都是默认用https来clone的
```
git clone https://github.com/xmake-io/xmake.git
```
但国内https访问github必须在shell中设置代理才能工作，换成git ssh之后不需要开代理也能工作：
```
git clone git@github.com:xmake-io/xmake.git
```
怎样让xmake update和xrepo默认走ssh而不是https？

**Comment by waruqi** [comment]
配下 pac.lua 替换下 url

https://xmake.io/#/zh-cn/package/remote_package?id=%e9%95%9c%e5%83%8f%e4%bb%a3%e7%90%86

**Reply by xq114** [reply]
本地测试pulling repository的时候不会走mirror
```
pulling repository(xmake-repo): https://gitlab.com/tboox/xmake-repo.git to C:\Users\xq114\AppData\Local\.xmake\repositories\xmake-repo ..
```
没有进过mirror函数

**Reply by waruqi** [reply]
mirror 目前只对包下载生效，只有proxy 配置是完全全局的

**Reply by waruqi** [reply]
我 dev 上对 repo clone 也加上 mirror 了
